### PR TITLE
Add codegen test for array comparision opt

### LINKED ISSUE #62531: Missed optimization on array comparison
The godbolt link: https://godbolt.org/z/dc9o3x

I think this snippet should just return `true`:
```rust
pub fn compare() -> bool {
    let bytes = 12.5f32.to_ne_bytes();
    bytes == if cfg!(target_endian = "big") {
        [0x41, 0x48, 0x00, 0x00]
    } else {
        [0x00, 0x00, 0x48, 0x41]
    }
}
```

The generated asm with `opt-level=3`:
```asm
example::compare:
        push    rax
        mov     al, 1
        pop     rcx
        ret
```

### PR DESCRIPTION
**Pull request by tesuji**
Fixed since rust 1.55
closes #62531

### CODE REVIEW

File: tests/assembly/62531-array-cmp.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Ensure the asm for array comparisons is properly optimized on x86.
+
+//@ min-llvm-version: 12
```
Comment (nikic) [inline tests/assembly/62531-array-cmp.rs:3]: Drop this line -- no need to specify min-llvm-version if it's lower than the oldest supported.

**Review by nikic — COMMENTED** [review]
Can you please make this an IR test instead (tests/codegen)? There should be no need for an assembly test here. Then it will work on non-x86 as well.

**Comment by matthiaskrgr** [comment]
@bors r-
https://github.com/rust-lang/rust/pull/125344#issuecomment-2121328546

**Pull request merged**